src/module-server/plugins/npm-plugin.ts

```diff
@@ -133,31 +132,63 @@ const nodeResolve = async (id: string, root: string) => {
   if (!result && subPath === '.')
     result = resolveLegacy(pkgJson, { browser: false, fields: ['main'] });
 
-  if (!result) {
+  if (!result && !('exports' in pkgJson)) {
     const extensions = ['.js', '/index.js', '.cjs', '/index.cjs'];
+    // If this was not conditionally included, this would have infinite recursion
+    if (subPath !== '.') extensions.unshift('');
     for (const extension of extensions) {
       const path = normalize(join(pkgDir, subPath) + extension);
-      if (existsSync(path)) return { path, idWithVersion };
+      const stats = await fs.stat(path).catch(() => null);
+      if (stats) {
+        if (stats.isFile()) return { path, idWithVersion };
+        if (stats.isDirectory()) {
+          // If you import some-package/foo and foo is a folder with a package.json in it,
+          // resolve main fields from the package.json
+          const result = await resolveFromFolder(path, '.', packageName);
+          if (result) return { path: result.path, idWithVersion };
+        }
```

Comment:
Not sure how concerned we are with this in this project but I figured I'd flag this lint warning in case you didn't see it ☝️

Reply:
I think I might disable that warning, but thanks for flagging it